src/main/java/pkmncore/DBManager.java

```diff
@@ -0,0 +1,112 @@
+        this.dbURL = dbURL;
+        this.username = username;
+        this.password = password;
+        this.driver = driver;
+    }
+
+    public Connection getConnection() throws PokemonError, SQLException {
+        setDriver();
+        return connectToDB();
+    }
+
+    public void save(String name, String height, String[] abilities) throws PokemonError {
+        savePokemonDetails(name, height, abilities);
+    }
+
+    public List<List<String>> getPokemon() throws PokemonError {
+        return getAllPokemon();
+    }
+
+    private void savePokemonDetails(String name, String height, String[] abilities) throws PokemonError {
+        try {
+            Connection connection = getConnection();
+            Statement statement = connection.createStatement();
+            saveNameAndHeight(name, height, statement);
+            saveAbilites(name, abilities, statement);
+            closeConnections(connection, statement);
+        } catch (SQLException e) {
+            throw new PokemonError(name + " has already been caught!");
+        }
+    }
+
+    private List<List<String>> getAllPokemon() throws PokemonError {
+        try {
+            Connection connection = getConnection();
+            Statement statement = connection.createStatement();
+            String sql = "SELECT * FROM POKEMON;";
+            List<List<String>> pokemon = listPokemon(connection, statement.executeQuery(sql));
+            closeConnections(connection, statement);
+            return pokemon;
+        } catch (SQLException e) {
+            throw new PokemonError("Something is wrong with the database!");
+        }
+    }
+
+    private List<List<String>> listPokemon(Connection connection, ResultSet caughtPokemon) throws SQLException {
+        List<List<String>> allPokemon = new ArrayList<List<String>>();
+        while (caughtPokemon.next()) {
+            List<String> pokemon = new ArrayList<String>();
+            pokemon.add(caughtPokemon.getString("name"));
+            pokemon.add(caughtPokemon.getString("height"));
+            getAbilities(pokemon, connection);
+            allPokemon.add(pokemon);
+        }
+        return allPokemon;
+    }
+
+    private void getAbilities(List<String> pokemon, Connection connection) throws SQLException {
+        Statement statement = connection.createStatement();
+        String sql = "SELECT * FROM ABILITIES WHERE (name) = '" + pokemon.get(0) + "';";
+        ResultSet abilities = statement.executeQuery(sql);
+        listAbilities(pokemon, abilities);
+    }
+
+    private void listAbilities(List<String> pokemon, ResultSet abilities) throws SQLException {
+        while (abilities.next()) {
+            pokemon.add(abilities.getString("ability"));
+        }
+    }
+
+    private void setDriver() throws PokemonError {
+        try {
+            Class.forName(driver);
+        } catch (ClassNotFoundException e) {
+            throw new PokemonError("Invalid driver");
```

Comment:
The exception would be more expressive it was called something along the lines of "InvalidDriverAppology" and probably took the actual driver name as an argument.

Otherwise its just a wrapper without a real purpose. It's not adding any extra information or value...